### PR TITLE
Gitlab CI Build

### PR DESCRIPTION
**Pull request by dmlond**
This is part 1 of a multi-part merge of the components required to support gitlab ci builds of the DDS codebase in Openshift, and a number of bots.

Note, the scripts in bin/ci/ are maintained in an internal [gitlab repository](https://gitlab.oit.duke.edu/ori-rad/openshift_deployment_pipeline), with Bats unit tests.

### CODE REVIEW

File: GitlabCI.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+The Duke Dataservice is making use of GitlabCI to augment the standard
+build and deployment process currently provided  by CircleCI. The
+GitlabCI platform allows us to test different processes to monitor the
+quality of our code, such as linting and code coverage, that do easily
+not fit into the CircleCI structure.
+
+#### Gitlab Github Mirroring
+For this to work, a project must be created in a Gitlab host that
+supports GitHub mirroring (this is an Enterprise feature). The project
+must mirror the github owner/repo to a GitHub owner/repo. **NOTE**
```
Comment (dmann) [inline GitlabCI.md]: Should one of these `github`s be a `gitlab`?

File: openshift/build_parameters [hunk 2]
```diff
@@ -0,0 +1,7 @@
+APPLICATION_NAME=duke-data-service
+GIT_REPO_URL=https://github.com/dmlond/duke-data-service.git
```
Comment (dmann) [inline openshift/build_parameters]: s/dmlond/Duke-Translational-Bioinformatics/?

**Review by dmann — CHANGES_REQUESTED** [review]
Looks good. Just a couple minor changes.

File: GitlabCI.md (hunk 1 above)
Comment (dmlond) [reply GitlabCI.md]: good catch.

File: openshift/build_parameters (hunk 2 above)
Comment (dmlond) [reply openshift/build_parameters]: I can do that, but this is really just an example to be modified as needed for testing. For example, the GIT_REF and APPLICATION_ENVIRONMENT would almost invariably need to be changed during manual testing for different git branches/tags

**Review by dmann — APPROVED** [review]

**Pull request merged**